### PR TITLE
Update fanout deploy yaml file to support using 2024 image

### PR DESCRIPTION
**Pull request by echuawu**
### Description of PR
Update fanout deploy yaml file to support using 2024 image in fanout switch

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [x] 202311
- [x] 202405

#### What is the motivation for this PR?
Use 2024 image in sonic fanout
#### How did you do it?
Update the fanout_sonic.yml
#### How did you verify/test it?
Run it in internal regression

### CODE REVIEW

File: ansible/roles/fanout/tasks/fanout_sonic.yml [hunk 1]
```diff
@@ -58,4 +58,4 @@
       include_tasks:
         sonic/fanout_sonic_202311.yml
       when: dry_run is not defined and incremental is not defined
-  when: "'2023' in fanout_sonic_version['build_version']"
+  when: "'2023' in fanout_sonic_version['build_version'] or '2024' in fanout_sonic_version['build_version']"
```
Comment (xwjiang-ms) [inline ansible/roles/fanout/tasks/fanout_sonic.yml:61]: Why does 2024 image use sonic/fanout_sonic_202311.yml?

Comment (echuawu) [reply ansible/roles/fanout/tasks/fanout_sonic.yml:61]: Currently, the sonic fanout configuration for 202311 and 202405 is the same. So use the same yaml file to handle the two branch.

Comment (xwjiang-ms) [reply ansible/roles/fanout/tasks/fanout_sonic.yml:61]: It doesn't make sense to me, if we follow this approach, we will be forced to patch `fanout_sonic_202311.yml` in the future. As time goes on, the number of patches will increase, raising the risk of importing new issues.
So I suggest to create a separate file for 202405.

Comment (echuawu) [reply ansible/roles/fanout/tasks/fanout_sonic.yml:61]: Reasonable, sure, I will have an internal discussion and update then.

Comment (echuawu) [reply ansible/roles/fanout/tasks/fanout_sonic.yml]: Hi @xwjiang-ms , updated the PR.

**Comment by bingwang-ms** [comment]
@xwjiang-ms Can you help review again?

File: ansible/roles/fanout/tasks/sonic/fanout_sonic_202405.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+fanout_sonic_202311.yml
```
Comment (xwjiang-ms) [inline ansible/roles/fanout/tasks/sonic/fanout_sonic_202405.yml:1]: Don't think we should inherit all fanout_sonic_202311.yml for 202405, although the code maybe the same, we can use cherry-pick to sync

Comment (echuawu) [reply ansible/roles/fanout/tasks/sonic/fanout_sonic_202405.yml:1]: Hi @nhe-NV , Do you have some comment?

Comment (nhe-NV) [reply ansible/roles/fanout/tasks/sonic/fanout_sonic_202405.yml:1]: Actaully I don't think duplicate code is a good option, at this moment, the content are identical, why we cna use the same file, or use the soft link to point to the same content.  In the future if we have to differeciate the content in different branch, then we can split, before that, I think it current implement is better.

Comment (wangxin) [reply ansible/roles/fanout/tasks/sonic/fanout_sonic_202405.yml:1]: I think it's OK to use soft link so far.
In the future, if 202405 and 202311 are getting different, the fix would be pretty easy.

**Review by wangxin — APPROVED** [review]

**Pull request merged**